### DISCUSSION TITLE
filter stack traces

### DISCUSSION THREAD

**Discussion by petergaultney** [pythonspeed/filprofiler#290] [Ideas]
If it wouldn't be too much work, I'd love to see some options to filter down the SVG flamegraph to eliminate segments of the stack that are irrelevant. In my programs, you're often 20 frames deep before you get to anything I've written, and everything above that is some kind of framework-ish thing that is completely irrelevant to memory usage.

Even something simple like accepting multiple regexes where any matched frame is elided would be amazing.

**Comment by itamarst** [comment]
That's a good idea. I've had some ideas about ways to automate this (basically so long the stack of frames is the same or almost the same length, most of the stack adds no information, so it should be possible to heuristically detect these and have them hidden by default, so long as there's a button to make them visible when necessary). Not something I will get to immediately, but hopefully at some point.

As an interim workaround: in the output folder, there's a `peak-memory.prof` file. The text format is fairly straightforward (lines in the form `frame1;frame2;frame3 count`), so you can write a little program to filter it. You can then install https://github.com/jonhoo/inferno, and render the SVG like so:

```shell-session
$ cat peak-memory.prof | inferno-flamegraph -i --colordiffusion
```

**Reply by itamarst** [reply]
(If you don't have the setup for installing Rust packages, you can also use the `flamegraph.pl` perl script from https://github.com/brendangregg/FlameGraph, it'll just have uglier output).
